web-common/src/features/entity-management/WatchFilesClient.ts

```diff
@@ -0,0 +1,79 @@
+import { fileArtifacts } from "@rilldata/web-common/features/entity-management/file-artifacts";
+import {
+  getRuntimeServiceGetFileQueryKey,
+  getRuntimeServiceIssueDevJWTQueryKey,
+  getRuntimeServiceListFilesQueryKey,
+  V1FileEvent,
+  V1WatchFilesResponse,
+} from "@rilldata/web-common/runtime-client";
+import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+import { WatchRequestClient } from "@rilldata/web-common/runtime-client/watch-request-client";
+import { get } from "svelte/store";
+import { queryClient } from "@rilldata/web-common/lib/svelte-query/globalQueryClient";
+
+export class WatchFilesClient {
+  private readonly client: WatchRequestClient<V1WatchFilesResponse>;
+  private seenFiles = new Set<string>();
+
+  public constructor() {
+    this.client = new WatchRequestClient<V1WatchFilesResponse>();
+    this.client.on("response", (res) => this.handleWatchFileResponse(res));
+    this.client.on("reconnect", () => this.invalidateAllFiles());
+  }
+
+  public static New() {
+    const watcher = new WatchFilesClient();
+    return watcher.client;
+  }
```

Comment:
This seems unconventional in Javascript. Can we just use the constructor to instantiate a new class?